src/http.rs

```diff
@@ -61,3 +72,87 @@ fn add_auth_headers<'a>(headers: &'a mut HeaderMap, user: &GlobalUser) {
+// detail to error messages, if desired.
+pub enum ErrorCodeDetail {
+    None,
+    WorkersKV,
+}
+
+// Format errors from the cloudflare-rs cli for printing.
+pub fn format_error(e: ApiFailure, err_type: ErrorCodeDetail) -> String {
+    match e {
+        ApiFailure::Error(status, api_errors) => {
+            print_status_code_context(status);
+            let mut complete_err = "".to_string();
+            for error in api_errors.errors {
+                let error_msg =
+                    format!("{} Error {}: {}\n", emoji::WARN, error.code, error.message);
+
+                let suggestion = match err_type {
+                    ErrorCodeDetail::WorkersKV => kv_help(error.code),
+                    _ => "",
+                };
+                if !suggestion.is_empty() {
+                    let help_msg = format!("{} {}\n", emoji::SLEUTH, suggestion);
+                    complete_err.push_str(&format!("{}{}", error_msg, help_msg));
+                } else {
+                    complete_err.push_str(&error_msg)
+                }
+            }
+            complete_err
+        }
+        ApiFailure::Invalid(reqwest_err) => format!("{} Error: {}", emoji::WARN, reqwest_err),
+    }
+}
+
+// For handling cases where the API gateway returns errors via HTTP status codes
+// (no API-specific, more granular error code is given).
+fn print_status_code_context(status_code: StatusCode) {
+    match status_code {
+        // Folks should never hit PAYLOAD_TOO_LARGE, given that Wrangler ensures that bulk file uploads
+        // are max ~50 MB in size. This case is handled anyways out of an abundance of caution.
+        StatusCode::PAYLOAD_TOO_LARGE =>
+            message::warn("Returned status code 413, Payload Too Large. Please make sure your upload is less than 100MB in size"),
+        StatusCode::GATEWAY_TIMEOUT =>
+            message::warn("Returned status code 504, Gateway Timeout. Please try again in a few seconds"),
+        _ => (),
+    }
+}
+
+// kv_help() provides more detailed explanations of Workers KV API error codes.
+// See https://api.cloudflare.com/#workers-kv-namespace-errors for details.
+fn kv_help(error_code: u16) -> &'static str {
+    match error_code {
+        7003 | 7000 => {
+            "Your wrangler.toml is likely missing the field \"account_id\", which is required to write to Workers KV."
+        }
+        // namespace errors
+        10010 | 10011 | 10012 | 10013 | 10014 | 10018 => {
+            "Run `wrangler kv:namespace list` to see your existing namespaces with IDs"
+        }
+        10009 => "Run `wrangler kv:key list` to see your existing keys", // key errors
+        // TODO: link to more info
+        // limit errors
+        10022 | 10024 | 10030 => "See documentation",
+        // TODO: link to tool for this?
+        // legacy namespace errors
+        10021 | 10035 | 10038 => "Consider moving this namespace",
```

Comment:
I know this isn't part of this PR, but do we know exactly what this means and if we can give them a better error message? This doesn't quite make sense to me

Reply:
for reference, the actual error codes for KV are excellent, and tell exactly what the problem is, so we print them to the console first. the messages included here are supplementary to give people actionable suggestions for how to fix their problems, they are _not_ the complete error code.

this is another reason why having separate functions for handling errors per module is useful: the messages that come out of the workers api are rather terse by comparison, so i would want to provide more granular handling there.